docs/adrs/0002-dedicated-timdex-record-field-methods.md

```diff
@@ -0,0 +1,101 @@
+
+Acknowledging this approach has been very successful for sources to date, it has also presented some issues:
+
+  * there are some hardcoded assumptions about the relationship of unique identifiers to accessible source URLs defined in methods not designed to be overridden
+  * there is an emerging pattern (as mentioned above) to breakout complex logic into methods, but is not consistently applied in naming or approach across sources
+  * extending a base transformers can be tricky if the `get_optional_fields()` is complex or reused variables across fields
+  * documentation for handling a particular field, unless broken out into a standalone method, was confined to inline comments
+  * testing transformations requires testing the full, final outputted record that `get_optional_fields()` contributes to 
+
+Additionally, there has been a desire to document how a source is mapped from the original "raw" record into a TIMDEX record.  The `get_optional_fields()` approach does not lend itself well to documentation as it relies heavily on inline comments and/or ad-hoc methods to breakout complex logic.  If every transformation had methods dedicated to particular fields, thereby providing a space for docstrings to explain the logic of that field handling, this might eventually support programmatic documentation of how fields are mapped for a source.
+
+Moving to a method-per-field approach will also allow for more granular logging.
+
+## Decision
+
+### 1- Refactor the base `Transformer` class to have all `TimdexRecord` fields as dedicated methods
+
+At the time of writing, the `TimdexRecord` has 30 fields:
+
+  * REQUIRED
+    * `source`
+    * `source_link`
+    * `timdex_record_id`
+    * `title`
+
+  * OPTIONAL
+    * `alternate_titles`
+    * `call_numbers`
+    * `citation`
+    * `content_type`
+    * `contents`
+    * `contributors`
+    * `dates`
+    * `edition`
+    * `file_formats`
+    * `format`
+    * `funding_information`
+    * `holdings`
+    * `identifiers`
+    * `languages`
+    * `links`
+    * `literary_form`
+    * `locations`
+    * `notes`
+    * `numbering`
+    * `physical_description`
+    * `publication_frequency`
+    * `publication_information`
+    * `related_items`
+    * `rights`
+    * `subjects`
+    * `summary`
+
+The base `Transformer` class will have a dedicated method for each of these 30 `TimdexRecord` fields, providing a clear and modular approach to defining logic for setting those fields. 
+
+The required fields will be an `@abstractmethod`, meaning any extending classes will be required to define them.
+
+The rest of the optional fields will receive an `@optional_field("<FIELD_NAME>")` decorator that will identify them as a) methods meant to parsing data for a given field, and b) the name of that field.  Using a decorator provides a few of benefits:
```

Comment:
Is `@optional_field` a custom decorator?

Reply:
Yep, example here: https://gist.github.com/ghukill/101ff97c1c892526c6ad62ee78c022fa#file-transmogrifier_field_methods-py-L30-L36